src/acl/AnnotateClient.cc

```diff
@@ -26,6 +26,7 @@ Acl::AnnotateClientCheck::match(ACLChecklist * const ch)
             tdata->annotate(request->notes(), &delimiters.value, checklist->al);
         return 1;
     }
+    debugs(28, 7, "fails: client has gone");
```

Comment:
This ACL is documented as 'always matches', though it may return 0. Probably this 'always' may be interpreted as 'while the transaction is still on' - so if the client is gone, there is no transaction and hence this ACL is not applicable.

Reply:
```
	If you want retrievals to always continue if they are being
	cached set 'quick_abort_min' to '-1 KB'.
```

> so if the client is gone, there is no transaction 

I am worried that the above assertion is false: I suspect it is possible to delete (i.e. invalidate) ClientStateData object while, say, HttpStateData part of the transaction is still ongoing. For example, can the documented `quick_abort_min -1 KB` setting have this effect (when client disappears early in a large download transaction)? If it can, when does Squid make the corresponding access.log record (e.g., after client disappears or after that large download completes)?

Let's confirm whether a "transaction" (for some definition of that vague term) may exist beyond ClientStateData lifetime before finalizing this PR.

> hence this ACL is not applicable.

If admins cannot control ClientStateData validity period during transaction lifetime, then they cannot (without heroid efforts) write reliable ACL-driven rules using an ACL which outcome depends on ClientStateData validity period:

```
# XXX: May randomly fail.
http_reply_access deny accessToBannedSite markClientAsBad
```

Reply:
> does Squid make the corresponding access.log record (e.g., after client disappears or after that large download completes)?

In my test, Squid logged just after client disappears (~ClientHttpRequest initiates logging with ERR_CLIENT_GONE  and then ~ConnStateData is called). The large download continues, as expected, honoring the `quick_abort_min -1 KB` setting.

>  confirm whether a "transaction" (for some definition of that vague term) may exist beyond ClientStateData lifetime

In order to confirm this we should resolve the 'vagueness' of this term, I think. For example, if the transaction ends just after it has been logged, then it does not exist after ~ConnStateData. Or we may decide that it does not  necessarily end after logging, i.e., continues as long as some artifacts/objects that it has initiated exist (such as the objects involved in storing the rest of response).